androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/SurveyDetailScreen.kt

```diff
@@ -50,8 +51,13 @@ fun SurveyDetailScreen(
     }
     var shouldShowExitConfirmationDialog by remember { mutableStateOf(false) }
 
+    LaunchedEffect(surveyDetailViewState.survey) {
+        surveyDetailViewState.survey?.let(surveyQuestionViewModel::updateStateWith)
+    }
```

Comment:
@AVI5HEK Can you remind me why are we observing stat like this?

Reply:
@kaungkhantsoe  due to the way iOS implemented the passing of the `survey` from one screen to another. Perhaps @phongvhd93  can shed some light on this.

Reply:
When a user taps on the `Start Survey` we will fetch the survey detail and pass it to the view model of the next screen. If there is an error, the user will stay at the `Survey Detail` screen. We won't fetch questions in the question screen, so passing the survey is required.

Reply:
@phongvhd93 The thing is if I comment that out it still works for questions but it won't submit the surveys.

Reply:
@kaungkhantsoe that's correct. We need the other VM for submitting the survey answers.